src/port1.0/portlint.tcl

```diff
@@ -115,6 +116,132 @@ proc portlint::seems_utf8 {str} {
     return true
 }
 
+# lint_checksum_types
+#
+# Given a list of checksum types, return a list of strings which are warnings
+# about deprecated checksum types, or missing recommended types.
+#
+# Returns an empty list if no issues are found.
+proc portlint::lint_checksum_type_list {types} {
+    set issues [list]
+
+    foreach type $types {
+        if {$type ni [portchecksum::list_preferred_checksum_types]} {
+            lappend issues "checksum type is deprecated: $type"
```

Comment:
I think it is. SHA1 is no longer secure.

Reply:
Agreed.  I had the impression we wanted to warn when any deprecated insecure algorithm was used in any case.

Reply:
No, we want to warn when secure algorithms are _not_ being used. Adding something else to the mix on top of the recommended ones _increases_ certainty that the file is correct. It's literally the opposite of a problem.

Reply:
Makes sense.  Will fix.

Reply:
OK, now deprecation warnings only occur if no recommended checksum types have been seen.

Reply:
> No, we want to warn when secure algorithms are not being used. Adding something else to the mix on top of the recommended ones increases certainty that the file is correct. It's literally the opposite of a problem

I think it's best not to encourage use of obsolete algorithms for human factors reasons because (as I've noted) SHA1 is not secure. Yes, in the context of using other secure hashes it can't cause much damage, but we don't want to encourage people to continue using the algorithm at all.

Reply:
It's not a valid use because you can't trust the upstream md5 or sha1 anyway so who cares if they advertise one. An md5 is especially easy to forge at this point, and a sha1 is getting there.

Reply:
(And if someone is still advertising the md5 of their package, that probably means they haven't been paying attention for years and years and the thing probably isn't maintained, which is another issue.)

Reply:
I'd say it's still useful to compare with upstream checksums. We don't have a lot of control over what upstream chooses to advertise (and believe it or not, some projects are actively worked on  without being run by people with understanding of which hashes are still good or not).